VDKQueue.m

```diff
@@ -422,6 +422,23 @@ - (void) removeAllPaths
 }
 
 
+- (void) stopWatching
+{
+    // This method must be called if we want this object to ever be dealloc'd. This is because detachNewThreadSelector:toTarget:withObject:
+    // retains the target (in this case, self), setting the retainCount to 2. Aside from the memory leak issue, if the thread is never terminated,
+    // the watcher thread might still try to send messages to a nonexistent delegate, causing your app to crash.
+    @synchronized(self)
+    {
+        // Shut down the thread that's scanning for kQueue events
+        kevent(_coreQueueFD, &(struct kevent){0, EVFILT_USER, EV_ENABLE, NOTE_TRIGGER, 0, NULL}, 1, NULL, 0, &(struct timespec){0,0});
```

Comment:
I could be wrong, but I think that `EVFILT_USER` is not enough to assert that the event was created by us: it could be a different app also sending an `EVFILT_USER` event. So we'll want a custom udata containing `[[NSProcessInfo processInfo] processIdentifier]` instead of NULL, so that we can compare it with our own pid before shutting down the thread.

Reply:
You can't send EVFILT_USER events to a different process. That would be some crazy easy interprocess communication. (Just as a sanity check I just tested this with two separate processes and verified you can't send kqueue user events from one to the other.) Even in the same process when you post a EVFILT_USER event you have to post it to a specific kqueue, it doesn't broadcast it to every thread waiting for a kevent.